src/components/app/data/services/course.js

```diff
@@ -17,11 +19,10 @@ export async function fetchCourseMetadata(enterpriseId, courseKey, options = {})
     const response = await getAuthenticatedHttpClient().get(url);
     return camelCaseObject(response.data);
   } catch (error) {
-    const errorResponseStatusCode = getErrorResponseStatusCode(error);
-    if (errorResponseStatusCode === 404) {
-      return null;
+    if (getErrorResponseStatusCode(error) !== 404) {
```

Comment:
[inform] Same general treatment for all API calls made during route loaders that have an expected error like 404.

That is, if the error is expected (e.g., known 404 as in this case), return valid data albeit empty. If the error is NOT expected (e.g., 500), still return valid/empty data but ensure the error is logged for observability,.